fedimint-server/src/config/mod.rs

```diff
@@ -471,9 +472,7 @@ impl ServerConfig {
         // initially, where we consider "module as a code" as "module as an instance at
         // runtime"
         let null_config_gen = ConfigGenParams::null();
-        for (module_instance_id, (kind, gen)) in registry.legacy_init_order_iter().enumerate() {
```

Comment:
Had to change this so that we can test `distributed_gen` without modules.  Long-term we need to move towards this for modularization anyway.

`MODULE_INSTANCE_ID_DKG_DONE` is introduced because without modules serving to synchronize the global messages, we could send a done message before DKG completes.